### PR TITLE
Add themetoggle

### PR DESCRIPTION
**Pull request by ggtylerr**
Adds a cool theme switcher:

https://user-images.githubusercontent.com/28498877/231881812-bf11f9f4-10be-4db4-bec9-6eb06c31a0f0.mp4

---

- Also moves the download link to the right (#6)
- Style code is messy, I know, lol
- ~~Moving between pages / sites causes it to briefly load the initial color scheme~~
  - ~~This is because the script loads after the page content does and I have no idea how to fix that~~
  - This is fixed in the 6th commit here
- Also the theme switcher is on the bottom right to align with the other icons. Tried it with the top right and it looked ugly. It can be moved pretty easily so it isn't too much of a problem.

### CODE REVIEW

**Review by lewisakura — CHANGES_REQUESTED** [review]
Can you please fix the colors? The purple accent appear to be wrong and it seems like at least one of the background colors is incorrect (see how the inline code looks on the download page).

**Comment by ggtylerr** [comment]
Yeah there shouldn't be any color changes, I just moved everything to different variables

File: src/components/Footer.astro [hunk 1]
```diff
@@ -31,6 +32,10 @@ import { Icon } from "astro-icon";
             <a href="/support" title="Support Vencord's Development">
                 <Icon name="fa-regular:heart" />
             </a>
+            <Theme client:load>
```
Comment (Vendicated) [inline src/components/Footer.astro:35]: this component would be better named ThemeToggle

File: src/components/Footer.astro [hunk 2]
```diff
@@ -31,6 +32,10 @@ import { Icon } from "astro-icon";
...
+                <Icon name="fa-regular:sun" slot="light"/>
```
Comment (Vendicated) [inline src/components/Footer.astro:36]: these Icons should have a title indicating their functionality! 
```suggestion
                <Icon name="fa-regular:sun" slot="light" title="Switch to dark theme" />
```

File: src/components/Theme.svelte [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        });
+    }
+
+    function themeSwitch() {
+        selected.update(x => x = (x === "Light") ? "Dark" : "Light");
+    }
+</script>
+
+<!-- Probably not the best way to go about this tbh -->
+<button on:click={themeSwitch} title="Switch to Dark">
```
Comment (Vendicated) [inline src/components/Theme.svelte:41]: this title can be removed if you add titles to the Icons themselves like mentioned above (and is wrong :P)
```suggestion
<button on:click={themeSwitch}>
```

File: src/components/Theme.svelte [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    {#if $selected === "Light"}
+        <slot name="light"/>
+    {:else if $selected === "Dark"}
+        <slot name="dark"/>
+    {/if}
+</button>
+
+<style>
+    button {
+        all: unset;
```
Comment (Vendicated) [inline src/components/Theme.svelte:51]: this removes the focus outline (see screenshot) which is important for people using keyboard navigation
image

Please add it back on button:focus

File: src/layouts/Layout.astro [hunk 5]
```diff
@@ -45,6 +45,15 @@ const { title, description, opengraph } = Astro.props;
         <link rel="sitemap" href="/sitemap-index.xml" />
     </head>
     <body>
+        <!-- Loads theme (this fixes a "flickering" bug 
+            that occurs when the theme script loads after
+            the page content does, causing it to briefly
+            show the original colors.) -->
+        <script is:raw>
+            const theme = localStorage.theme.toLowerCase();
```
Comment (Vendicated) [inline src/layouts/Layout.astro:53]: This will error if not set! 
```suggestion
            const theme = localStorage.theme?.toLowerCase();
```

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: src/components/Footer.astro (hunk 2 above)
Comment (ggtylerr) [reply src/components/Footer.astro]: Thanks for pointing that out, I completely forgot that this (and it also made me realize that it was always showing the current theme's icon and not the one it was going to switch to)

File: src/components/ThemeToggle.svelte [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    const options = ["Light", "Dark"] as const;
+
+    const initialValue = IS_SERVER
+        ? "Dark"
+        : (() => {
+              const stored = localStorage.theme;
+              if (stored && options.includes(stored)) return stored;
+
+              if (window.matchMedia("(prefers-color-scheme: light").matches)
```
Comment (Vendicated) [inline src/components/ThemeToggle.svelte:14]:
```suggestion
              if (window.matchMedia("(prefers-color-scheme: light)").matches)
```

but funnily enough this works even without closing it, i love html <3

**Review by Vendicated — APPROVED** [review]

**Pull request merged**